src/pages/index.js

```diff
@@ -4,73 +4,197 @@ import Section from "../components/Section.js";
+      handleImageClick,
+      handleCardDeleteClick,
+      handleLikeClick
+    );
+    return cardElement.getView();
+  }
 
-cardSection.renderItems(constants.initialCards);
+  const profileEditForm = new PopupWithForm("#profile-edit-popup", (data) => {
+    const { header, description } = data;
+    profileEditForm.renderLoading(true);
+
+    api
+      .updateUserInfo(header, description)
+      .then((res) => {
+        userInformation.setUserInfo(res);
+        profileEditForm.close();
+        profileEditForm.resetForm();
+        profileEditValidator.disableButton();
+      })
+      .catch((err) => {
+        console.error("Error updating profile:", err);
+      })
+      .finally(() => {
+        profileEditForm.renderLoading(false);
+      });
+  });
+
+  constants.profileEditButton.addEventListener("click", () => {
+    profileEditValidator.resetValidation();
+    const userInfo = userInformation.getUserInfo();
+    constants.profileTitleInput.value = userInfo.title || "";
+    constants.profileDescriptionInput.value = userInfo.description || "";
+    profileEditForm.open();
+  });
+
+  profileEditForm.setEventListeners();
+
+  function handleAvatarFormSubmit(data) {
+    avatarImagePopup.renderLoading(true);
+
+    api
+      .updateAvatar(data.link)
+      .then((res) => {
+        userInformation.setUserAvatar(res);
+        avatarImagePopup.close();
+        avatarImagePopup.resetForm();
+        avatarValidator.disableButton();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        avatarImagePopup.renderLoading(false);
+      });
+  }
+
+  function handleCardDeleteClick(card) {
+    confirmDeletePopup.open();
+    confirmDeletePopup.setSubmitAction(() => {
+      return api
+        .deleteCard(card._id)
+        .then(() => {
+          confirmDeletePopup.close();
+          card.handleTrashButton();
+        })
+        .catch((err) => {
+          console.log(err);
+        });
+    });
+  }
+
+  function handleLikeClick(cardId, action) {
+    if (action === "dislike") {
+      return api.dislikeCard(cardId);
+    } else if (action === "like") {
+      return api.likeCard(cardId);
+    } else {
+      return Promise.reject(new Error("Invalid action"));
+    }
+  }
+
+  const addCardForm = new PopupWithForm("#add-card-popup", (data) => {
+    const { title, URL } = data;
+    addCardForm.renderLoading(true);
+    api
+      .addCard(title, URL)
+      .then((cardData) => {
+        cardSection.addItem(createCard(cardData));
+        addCardForm.close();
+        addCardForm.resetForm();
+        addCardValidator.disableButton();
+      })
+      .catch((err) => {
+        console.error("Error adding card:", err);
+      })
+      .finally(() => {
+        addCardForm.renderLoading(false);
+      });
+  });
+
+  addCardForm.setEventListeners();
+  confirmDeletePopup.setEventListeners();
+
+  constants.addCardButton.addEventListener("click", () => {
+    addCardForm.open();
```

Comment:
Please, make method `toggleButtonState` **public** by removing the low dash and call it on openings to work around the enabling the button on `finally`


```js

    addCardValidator.toggleButtonState();

```